src/router/index.js

```diff
@@ -38,8 +40,27 @@ let router = new Router({
     },
     {
       path: '/profile',
-      name: 'Home',
+      name: 'Profile',
       component: Profile,
+      meta: {
+        requiresAuth: true,
+      },
+      children: [
+        { path: ':id', component: ProfileDetail },
+        {
+          path: ':id/edit',
+          component: ProfileEdit,
+          beforeEnter (to, from, next) {
+            const routeId = to.params.id
```

Comment:
this is the userId yah? perhaps can name it userId vs routeId

Reply:
True but it is the userId in the route, as opposed to the userId of the current user which could be different if someone is trying to edit someone elses profile.  Just tried to keep them seperate as much as possible but I'm not stuck to this naming convention so if you have any suggestions I can go with it.